docs/rsdl/rapid-pro-rsdl-abnf.md

```diff
@@ -0,0 +1,87 @@
+# RAPID Pro syntax
+
+July 2020
+
+## Overview
+
+This grammar uses ABNF as defined by [RFC5234](https://tools.ietf.org/html/rfc5234).
+
+## Syntax rules
+
+- [Model](#model)
+- [Type definition](#type-definition)
+- [Enum definition](#enum-definition)
+- [Service definition](#service-definition)
+- [Core syntax elements](#core-syntax-elements)
+
+### Model
+
+```ABNF
+model               = 1*modelElement
+
+modelElement        = typeDefinition /
+                      serviceDefinition /
+                      enumDefinition
+```
+
+### Type definition
+
+```ABNF
```

Comment:
Out of curiosity. Could this be generated at some point based on the code rules?

Reply:
By "This" I assume you mean the ABNF. I think that would be an interesting project but more work than is justified for just that one language. The parser framework that is used (https://github.com/datalust/superpower) is quite declarative and it might be possible to translate the syntax rules into ABNF. The parser framework distinguishes between lexing and parsing for the the same reasons that is done in manually coded parsers. The Lexer is not written in a declarative manner which might make the ABNF incomplete. 
But nonetheless, the ABNF, at this point was the relatively straightforward. Writing the transformation document took  longer.